Update using generic (#498)
jakartaee/data: spec/src/main/asciidoc/repository.asciidoc

[source,java]
----

> PageRequest<Product> pageRequest = PageRequest.ofSize(20).page(1).sortBy(Sort.desc("price"));

Reviewer: ```suggestion
PageRequest<Product> pageRequest = PageRequest.of(Product.class).size(20).page(1).sortBy(Sort.desc("price"));
```

@Repository
public interface CustomerRepository extends BasicRepository<Customer, Long> {
  KeysetAwareSlice<Customer> findByZipcodeOrderByLastNameAscFirstNameAscIdAsc(
                                 int zipcode, PageRequest<Customer> pageRequest);
}
----

You can obtain the initial page relative to an offset and subsequent pages relative to the last entity of the current page as follows,

[source,java]
----

> for (PageRequest<Customer> p = PageRequest.ofSize(50); p != null; ) {

Reviewer: ```suggestion
for (PageRequest<Customer> p = PageRequest.of(Customer.class).size(50); p != null; ) {
```

[source,java]
----
Customer c = ...

> PageRequest<Customer> p = PageRequest.ofSize(50).afterKeyset(c.lastName, c.firstName, c.id);

Reviewer: ```suggestion
PageRequest<Customer> p = PageRequest.of(Customer.class).size(50).afterKeyset(c.lastName, c.firstName, c.id);
```

@Repository
public interface CustomerRepository extends BasicRepository<Customer, Long> {
  @Query("SELECT o FROM Customer o WHERE (o.totalSpent / o.totalPurchases > ?1)")
  KeysetAwareSlice<Customer> withAveragePurchaseAbove(float minimum, PageRequest<Customer> pageRequest);
}
----

Example traversal of pages:

[source,java]
----

> for (PageRequest<Customer> p = PageRequest.ofSize(25).sortBy(Sort.desc("yearBorn"),
>                                                       Sort.asc("name"),
>                                                       Sort.asc("id"));

Reviewer: ```suggestion
for (PageRequest<Customer> p = Order.by(_Customer.yearBorn.desc(),
                                        _Customer.name.asc(),
                                        _Customer.id.asc())
                                    .pageSize(25);
```

[source,java]
----
float priceMidpoint = 50.0f;

> PageRequest<Product> pageRequest = PageRequest.ofSize(10)
>                                         .sortBy(Sort.asc("price"), Sort.asc("id"))
>                                         .afterKeyset(priceMidpoint, 0L);

Reviewer: ```suggestion
PageRequest<Product> pageRequest = Order.by(_Product.price.asc(), _Product.id.asc())
                                        .pageSize(10)
                                        .afterKeyset(priceMidpoint, 0L);
```
